### PR TITLE
Update Uglify

### PR DESCRIPTION
**Pull request by thomas-huston-zocdoc**
@ZocDoc/polaris-devs 

Changes
--
- Bumped us up to the latest version of uglify. This will allow us to start generating source maps for minified JS code (to come in a future PR).

### CODE REVIEW

File: src/bundler.js [hunk 1]
```diff
@@ -72,6 +70,9 @@ var fs = require("fs"),
     cssValidator = require('./css-validator'),
     directoryCrawler = require('./directory-crawler'),
     sourceMap = require('./source-map-utility'),
+    tasks = {
+        minifyJs: require('./tasks/minify/minify-js')
```
Comment (thomas-huston-zocdoc) [inline src/bundler.js]: This file is getting pretty hard to work in - it's about 800 lines. Pulled the minification code out into a separate file. Extracting actions like this into small independent files will make it easier if we ever migrate over to gulp.

**Comment by thomas-huston-zocdoc** [comment]
GitHub won't show a lot of the diff here, so I'll comment on some individual commits for more context.

File: src/bundler.js (hunk 1 above)
Comment (Andrew-Hagedorn-ZocDoc) [inline src/bundler.js]: Its a mess.  Its definitely the right call to extract anything you touch.

Comment (Andrew-Hagedorn-ZocDoc) [inline src/bundler.js]: Plus, then you move into fast unit tests to cover most things.

**Pull request merged**